rm/rm-server/src/main/java/org/ow2/proactive/resourcemanager/core/RMCore.java

```diff
@@ -1431,30 +1431,36 @@ public BooleanWrapper createNodeSource(String nodeSourceName, String infrastruct
      */
     @Override
     public BooleanWrapper deployNodeSource(String nodeSourceName) {
-
         logger.info("Deploy node source " + nodeSourceName + REQUESTED_BY_STRING + this.caller.getName());
-
         if (!this.deployedNodeSources.containsKey(nodeSourceName)) {
-
             NodeSourceDescriptor nodeSourceDescriptor = this.getDefinedNodeSourceDescriptorOrFail(nodeSourceName);
+            this.updateNodeSourceDescriptorWithStatusAndPersist(nodeSourceDescriptor, NodeSourceStatus.NODES_DEPLOYED);
+            deployNodeSourceOrFail(nodeSourceName, nodeSourceDescriptor);
+        } else {
+            logger.debug("Node source " + nodeSourceName + " is already deployed");
```

Comment:
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Use already-defined constant 'NODE_SOURCE_STRING' instead of duplicating its value here. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS1192)